### PR TITLE
Fixes Lazarus injector has a small bug + Lazarus QoL

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Fixes Lazarus injector has a small bug
* revived mob's faction can be unintentionally changed (for example, carps will lose carp faction)
* revived hostile-mob without player occupation (but mind exists) will follow AI act to attack people
* QOL, ghost notification.

## Why It's Good For The Game
bug fix good

## Testing Photographs and Procedure

image

## Changelog
:cl:
fix: Lazarus injector will not accidentally change faction status when it has a player mind.
fix: Mobs with player mind that are revived by Lazarus injector will have their AI status off, so that people will not be attacked randomly.
add: Lazarus injector will not revive a mob with an unrevivable mind. If it revives a mob, it will notify its ghost.
/:cl:

### CODE REVIEW

File: code/modules/mining/equipment/lazarus_injector.dm [hunk 1]
```diff
@@ -26,18 +26,21 @@
...
-						H.robust_searching = 1
-						H.friends += user
-						H.attack_same = 1
-						log_game("[key_name(user)] has revived hostile mob [key_name(target)] with a malfunctioning lazarus injector")
-					else
-						H.attack_same = 0
+				if(M.mind)
+					M.AIStatus = AI_OFF // don't let them attack people randomly after revived
+				else // only do this to mindless mobs
+					M.faction = list("neutral")
```
Comment (PowerfulBacon) [inline code/modules/mining/equipment/lazarus_injector.dm]: Why do we only want to set the faction for mindless mobs?

Comment (EvilDragonfiend) [reply code/modules/mining/equipment/lazarus_injector.dm]: yeah I think it's better to remove

**Review by Penwin0 — COMMENTED** [review]
Thought the faction being set to neutral thing was on purpose, to make them non hostile anymore

File: code/modules/mining/equipment/lazarus_injector.dm (hunk 1 above)
Comment (Penwin0) [reply code/modules/mining/equipment/lazarus_injector.dm]: I think this is intended behaviour, to make the simple mob non agressive, mindless doesn't mean it has no AI

**Comment by EvilDragonfiend** [comment]
So, what do I do?

**Review by Rukofamicom — APPROVED** [review]

**Pull request merged**